src/js/components/Calculator.js

```diff
@@ -0,0 +1,103 @@
+import Modifiers from '../components/Modifiers.js';
+import Digits from '../components/Digits.js';
+import Operations from '../components/Operations.js';
+import { isOperandFirst, isOperandFull } from '../common/validations.js';
+import { calculate, formatOperand } from '../common/utils.js';
+
+const initialState = {
+    total: '0',
+    leftOperand: '0',
+    rightOperand: null,
+    operation: null,
+};
+class Calculator {
+    constructor() {
+        this.state = initialState;
+        this.$digits = document.querySelector('div.digits');
+        this.$modifiers = document.querySelector('div.modifiers');
+        this.$operations = document.querySelector('div.operations');
+        this.$total = document.querySelector('#total');
+        this.initializeComponents();
+    }
```

Comment:
```suggestion
    state;
    $digits;
    $modifiers;
    $operations;
    $total;
    
    constructor() {
        this.state = { ...initialState };
        this.$digits = document.querySelector('div.digits');
        this.$modifiers = document.querySelector('div.modifiers');
        this.$operations = document.querySelector('div.operations');
        this.$total = document.querySelector('#total');
        this.initializeComponents();
    }
```
이렇게 클래스에서 사용되는 멤버변수를 미리 정의해주면 이 클래스가 어떤 멤버변수를 사용하는지 한 눈에 파악할 수 있어서 좋답니다!

그리고 initState를 state에 바로 할당할 경우 레퍼런스를 공유하기 때문에 예상하는 것 처럼 동작하지 않을 확률도 있답니다!